internal/adapters/entrypoints/rest/handlers/get_trusted_accounts_test.go

```diff
@@ -68,18 +70,21 @@ func TestNewGetTrustedAccountsHandler(t *testing.T) {
 		assert.Equal(t, "Test Account 2", response.Accounts[1].Name)
 		repo.AssertExpectations(t)
 		hashMock.AssertExpectations(t)
+		signerMock.AssertExpectations(t)
 	})
 	t.Run("should return 500 on error", func(t *testing.T) {
 		recorder := httptest.NewRecorder()
 		request := httptest.NewRequest("GET", "/management/trusted-accounts", nil)
 		repo := mocks.NewTrustedAccountRepositoryMock(t)
 		repo.On("GetAllTrustedAccounts", mock.Anything).Return(nil, errors.New("database error"))
 		hashMock := &mocks.HashMock{}
-		useCase := lpuc.NewGetTrustedAccountsUseCase(repo, hashMock.Hash)
+		signerMock := &mocks.SignerMock{}
+		useCase := lpuc.NewGetTrustedAccountsUseCase(repo, hashMock.Hash, signerMock)
 		handler := http.HandlerFunc(handlers.NewGetTrustedAccountsHandler(useCase))
 		handler.ServeHTTP(recorder, request)
 		assert.Equal(t, http.StatusInternalServerError, recorder.Code)
 		repo.AssertExpectations(t)
 		hashMock.AssertNotCalled(t, "Hash")
+		signerMock.AssertNotCalled(t, "Validate")
```

Comment:
I just noticed that the handler under test here is not handling all possible errors and is mapping all of them to a generic 500 one. Is this on purpose? Otherwise I think we should manage each error and sent a response based on the type of error and write tests accordingly.